zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -488,8 +488,7 @@ impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
         self.is_insufficient_balance
             .assign(region, offset, caller_balance, value)?;
 
-        let is_insufficient = value > caller_balance;
-
+        let is_insufficient = (value > caller_balance) && (is_call || is_callcode);
```

Comment:
The corresponding circuit expression only does `is_insufficient = value > caller_balance`.  This is because when `not(is_call || is_callcode)` then `value = 0` so `value > caller_balance == 0 > caller_balance == false`.

So I believe adding the `&& (is_call || is_callcode` here is redundant.  Nevertheless maybe it makes things more clear?  This is just an observation, not a request to change anything.

Reply:
i also prefer to remove the redundant `&& (is_call || is_callcode`.